### PR TITLE
Initial commit of existing connectors

### PR DESCRIPTION
**Pull request by twdsilva**
@karanmehta93 @joshelser @vincentpoon Can you please review?
I tried to clean up the pom to remove dependencies that are not required, but I didn't get all of them.

### CODE REVIEW

**Comment by vincentpoon** [comment]
@twdsilva  I get an error when compiling phoenix-spark:

[INFO] --- scala-maven-plugin:3.4.4:compile (scala-compile-first) @ phoenix-spark ---
[INFO] /Users/vincent.poon/git_public/phoenix-connectors/phoenix-spark/src/main/java:-1: info: compiling
[INFO] /Users/vincent.poon/git_public/phoenix-connectors/phoenix-spark/src/main/scala:-1: info: compiling
[INFO] Compiling 24 source files to /Users/vincent.poon/git_public/phoenix-connectors/phoenix-spark/target/classes at 1547778775727
Downloading from apache release: https://repository.apache.org/content/repositories/releases/jline/jline/2.12.1/jline-2.12.1.pom
Downloading from central: https://repo.maven.apache.org/maven2/jline/jline/2.12.1/jline-2.12.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/jline/jline/2.12.1/jline-2.12.1.pom (17 kB at 652 kB/s)
[ERROR] /Users/vincent.poon/git_public/phoenix-connectors/phoenix-spark/src/main/scala/org/apache/phoenix/spark/FilterExpressionCompiler.scala:30: error: too many arguments for method getDateFormatter: (x$1: String)java.text.Format
[ERROR]   val dateformatter:Format = DateUtil.getDateFormatter(DateUtil.DEFAULT_DATE_FORMAT, DateUtil.DEFAULT_TIME_ZONE_ID)
[ERROR]                                                       ^
[ERROR] /Users/vincent.poon/git_public/phoenix-connectors/phoenix-spark/src/main/scala/org/apache/phoenix/spark/FilterExpressionCompiler.scala:31: error: too many arguments for method getTimestampFormatter: (x$1: String)java.text.Format
[ERROR]   val timeformatter:Format = DateUtil.getTimestampFormatter(DateUtil.DEFAULT_TIME_FORMAT, DateUtil.DEFAULT_TIME_ZONE_ID)
[ERROR]

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+          <links>
+            <link>http://hbase.apache.org/apidocs/</link>
+          </links>
+        </configuration>
+        <executions>
+          <execution>
+            <id>attach-javadocs</id>
+            <goals>
+              <!-- TODO turn back on javadocs - disabled now for testing -->
+              <!-- <goal>jar</goal> -->
```
Comment (vincentpoon) [inline pom.xml]: Can turn this on.  Maybe only for releases, if there's a way to do that

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,778 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>2.4</version>
+        <executions>
+          <execution>
+            <phase>prepare-package
+            </phase>
+            <goals>
+              <goal>test-jar</goal>
```
Comment (vincentpoon) [inline pom.xml]: nit: this is creating a test-jar for the parent also

**Comment by twdsilva** [comment]
@vincentpoon  Did you do a mvn install with the head of the 4.x-HBase-1.4 branch of phoenix?

**Comment by vincentpoon** [comment]
@twdsilva I had some old version, with the latest it now works, thanks

**Pull request closed without merging**